rotkehlchen/exchanges/binance.py

```diff
@@ -680,6 +681,73 @@ def _query_margined_futures_balances(
 
         return balances
 
+    def _query_pools_balances(
+            self,
+            balances: DefaultDict[Asset, Balance],
+    ) -> DefaultDict[Asset, Balance]:
+
+        def process_pool_asset(asset_name: str, asset_amount: FVal) -> None:
+
+            if asset_amount == ZERO:
+                return None
+
+            try:
+                asset = asset_from_binance(asset_name)
+            except UnsupportedAsset as e:
+                self.msg_aggregator.add_warning(
+                    f'Found unsupported {self.name} asset {asset_name}. '
+                    f'Ignoring its margined futures balance query. {str(e)}',
+                )
+                return None
+            except UnknownAsset as e:
+                self.msg_aggregator.add_warning(
+                    f'Found unknown {self.name} asset {asset_name}. '
+                    f'Ignoring its margined futures balance query. {str(e)}',
+                )
+                return None
+            except DeserializationError as e:
+                log.error(
+                    f'{self.name} balance deserialization error ',
+                    f'with asset {asset_name}. {str(e)}',
+                )
+                return None
+
+            try:
+                usd_price = Inquirer().find_usd_price(asset)
+            except RemoteError as e:
+                self.msg_aggregator.add_error(
+                    f'Error processing {self.name} balance entry due to inability to '
+                    f'query USD price: {str(e)}. Skipping margined futures balance entry',
+                )
+                return None
+
+            balances[asset] += Balance(
```

Comment:
nitpick: I don't like inner functions in python and especially inner functions like this that form a closure. It's hard to understand what's happening and error prone because the relationship between function input and output is unclear since it captures the variables of the parent function. Please try to avoid them. When writing code the most important thing is to make it as clear as possible so that yourself in 3 months, or your co-workers can very easily understand what's happening.

Reply:
I had this is mind because you have told me before but since I had to repeat the same steps for two entries, I believed it was better to use a inner function like this following the recommendation in the FTX PR.